### PR TITLE
Ato 981 set is new account in new orch session start

### PR DESCRIPTION
**Pull request by Ryan-Andrews99**
## What
- Sets isNewAccount in the new Orch session to UNKNOWN at session start

## How to review
- Code review commit-by-commit 
- Deploy to a development environment 
    - Run through a jorney 
    - Check session now contains UNKNOWN in the isNewAccount field

## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/OrchSessionItem.java [hunk 1]
```diff
@@ -8,9 +8,18 @@
 public class OrchSessionItem {
 
     public static final String ATTRIBUTE_SESSION_ID = "SessionId";
+    public static final String ATTRIBUTE_IS_NEW_ACCOUNT = "IsNewAccount";
+
+    public enum AccountState {
```
Comment (billrdunn) [inline orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/OrchSessionItem.java:13]: Should this exist as a separate class in shared/entity?

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java [hunk 2]
```diff
@@ -77,6 +77,7 @@ public OrchSessionItem addOrUpdateSessionId(
                 newItem =
                         new OrchSessionItem()
                                 .withSessionId(newSessionId)
+                                .withAccountState(OrchSessionItem.AccountState.UNKNOWN)
```
Comment (billrdunn) [inline orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java:80]: Does this also need to be added to the `addSession` method above this one?

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/OrchSessionItem.java (hunk 1 above)
Comment (Ryan-Andrews99) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/OrchSessionItem.java:13]: Not really sure we'd reuse this anywhere else apart from _session-y_ contexts, so not sure what benefit we'd get from separating it out

File: integration-tests/src/test/java/uk/gov/di/authentication/services/OrchSessionServiceIntegrationTest.java [hunk 3]
```diff
@@ -35,7 +37,39 @@ void shouldReturnEmptyIfSessionNotFound() {
...
+        assertTrue(retrievedSession.isPresent());
+        assertThat(
+                retrievedSession.get().getIsNewAccount(),
+                equalTo(OrchSessionItem.AccountState.UNKNOWN));
+    }
+
+    @Test
+    void shouldUpdatePreviousSessionWithNewId() {
+        withPreviousExistingAccountSession();
+
```
[the remark this answers is not in the document]
Comment (Ryan-Andrews99) [reply integration-tests/src/test/java/uk/gov/di/authentication/services/OrchSessionServiceIntegrationTest.java:53]: I see what you mean, will add!

[the remark this answers is not in the document]
Comment (Ryan-Andrews99) [reply integration-tests/src/test/java/uk/gov/di/authentication/services/OrchSessionServiceIntegrationTest.java:55]: Have added!

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java (hunk 2 above)
Comment (Ryan-Andrews99) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java:80]: I wouldn't really expect `addSession` to modify the session object I pass in, but given it's only used in one place we could rename to `addNewSession` and do the refactoring, what do you think?

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/OrchSessionItem.java (hunk 1 above)
Comment (billrdunn) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/OrchSessionItem.java:13]: fair enough!

File: orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java (hunk 2 above)
Comment (billrdunn) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java:80]: Does `.withAccountState(OrchSessionItem.AccountState.UNKNOWN)` need adding [here](https://github.com/govuk-one-login/authentication-api/blob/d8f198ffffb1eb06e023ed35f879820ac940227b/oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java#L515)?

Comment (Ryan-Andrews99) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java:80]: I think on refactoring we should probably do something similar to what @GHSwallow has done here https://github.com/govuk-one-login/authentication-api/pull/5384, where the new session constructor should have some defaults like `isNewAccount = UNKNOWN` similar to the current session class. Thoughts?

Comment (billrdunn) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java:80]: yep that sounds like a good idea to me 👍

Comment (Ryan-Andrews99) [reply orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/OrchSessionService.java]: Cool have added this

**Review by billrdunn — APPROVED** [review]

**Pull request merged**